subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractConsoleFunctionalSpec.groovy

```diff
@@ -46,6 +46,24 @@ class AbstractConsoleFunctionalSpec extends AbstractIntegrationSpec {
         return styledString
     }
 
+    /**
+     * Grabs the grouped output of a task
+     * @param taskName a fully qualified task name e.g. :project:task1
+     * @return
+     */
+    protected String taskOutputFrom(String taskName) {
```

Comment:
Sorry for being unclear. The implementation here assumes all output exists within one "group" when it may be split between multiple (if task :a has grouped output, then :b, then :a). We may just need to change the regex matching to handle multiple groups of logs for the same task.